README.md

```diff
@@ -1,18 +1,41 @@
 # retry
 
-An expressive, flexible retry package for Go.
+A small retry package for Go.
 
-[![GoDoc](https://godoc.org/github.com/golang/gddo?status.svg)](https://godoc.org/go.coder.com/retry)
+[![GoDoc](https://godoc.org/github.com/golang/gddo?status.svg)](https://godoc.org/github.com/coder/retry)
 
-## Features
+```
+go get github.com/coder/retry
+```
 
-- Backoff helper
-- Retrying net.Listener wrapper
+## Features
+- For loop experience instead of closures
+- Only 4 exported methods
+- No external dependencies
 
 ## Examples
 
-See [retry_example_test.go](retry_example_test.go)
-
-## We're Hiring!
-
-If you're a passionate Go developer, send your resume and/or GitHub link to [jobs@coder.com](mailto:jobs@coder.com).
+Wait for connectivity to google.com, checking at most once every
+second.
+```go
+func pingGoogle(ctx context.Context) error {
+    r := retry.New(time.Second, time.Second*10)
+    for r.Wait(ctx) {
+        _, err := http.Get("https://google.com")
+        r.SetError(err)
```

Comment:
Why do we need a set error function? Couldn't we return if there's an error? If there's no error, the user would break.

What's the advantage over using a ticker? Is it the maximum retries?